flood_adapt/object_model/tipping_point.py

```diff
@@ -0,0 +1,338 @@
+        self.calculate_sea_level_at_threshold(tp_results)
+
+        tp_results["strategy"] = self.attrs.strategy
+        # SAVE
+        tp_results.drop(columns=["object"]).to_csv(
+            tp_path / "tipping_point_results.csv"
+        )
+
+    def scenario_has_run(self, scenario_obj):
+        # TODO: once has_run is refactored (external) we change below to make it more direct
+        for db_scenario, finished in zip(
+            Database().scenarios.list_objects()["objects"],
+            Database().scenarios.list_objects()["finished"],
+        ):
+            if scenario_obj == db_scenario and finished:
+                return True
+        return False
+
+    def check_tipping_point(self, scenario: Scenario):
+        """Load results and check if the tipping point is reached."""
+        info_df = pd.read_csv(
+            scenario.init_object_model().direct_impacts.results_path.joinpath(
+                f"Infometrics_{scenario.direct_impacts.name}.csv"
+            ),
+            index_col=0,
+        )
+        # add to self.scenarios the results of the value metric for the given tipping point metric
+        for metric in self.attrs.tipping_point_metric:
+            self.scenarios[scenario.attrs.name][f"{metric[0]}_value"] = info_df.loc[
+                metric[0], "Value"
+            ]
+
+        # if any tipping point is reached, return True
+        # TODO: maybe change to different approach if more than one tipping
+        # point is being assessed (instead of any, maybe you want to check
+        # which TPs are reached and return a dict with the results)
+        return any(
+            self.evaluate_tipping_point(
+                info_df.loc[metric[0], "Value"],
+                metric[1],
+                metric[2],
+            )
+            for metric in self.attrs.tipping_point_metric
+        )
+
+    def evaluate_tipping_point(self, current_value, threshold, operator):
+        """Compare current value with threshold for tipping point."""
+        operations = {"greater": lambda x, y: x >= y, "less": lambda x, y: x <= y}
+        return operations[operator](current_value, threshold)
+
+    def calculate_sea_level_at_threshold(self, tp_results):
+        for metric in self.attrs.tipping_point_metric:
+            metric_name, threshold, operator = metric
+            # Remove NaN values to prevent interpolation errors
+            valid_data = tp_results[["sea level", f"{metric_name}_value"]].dropna()
+
+            x = valid_data["sea level"].values
+            y = valid_data[f"{metric_name}_value"].values
+
+            # Create a linear interpolation function
+            interpolation_function = interp1d(y, x, fill_value="extrapolate")
+
+            tp_results[f"{metric_name}_ATP"] = interpolation_function(threshold)
+
+    # create a function that plots the results from tp_path / "tipping_point_results.csv" against the SLR values
+    def plot_results(self):
```

Comment:
the plot function is quite simple for now: have a linear function connecting different combinations of damage and sea level, and comparing with the tipping point thresholds. It generates a different plot for each tipping point metric. Any thoughts?

Reply:
![image](https://github.com/user-attachments/assets/1adfe3bc-4635-4b06-a60d-756de69ed85f)